tests/console/yast2_samba.pm

```diff
@@ -14,258 +14,338 @@ use strict;
+    # set admin password and search base dn
     wait_screen_change { send_key 'alt-a' };
-    for (1 .. 20) { send_key 'backspace'; }
-    type_string 'cn=admin';
-    wait_screen_change { send_key 'alt-l' };
+    type_string 'cn=admin,dc=qa,dc=suse,dc=de';
+    wait_screen_change { send_key 'alt-p' };
     type_string 'testing';
-    wait_screen_change { send_key 'alt-v' };
+    wait_screen_change { send_key 'alt-g' };
     type_string 'testing';
-    # use database as default for ldap client
-    wait_screen_change { send_key 'alt-u' };
-    send_key 'alt-n';
-    assert_screen 'yast2_ldap_configuration_kerberos';
-    wait_screen_change { send_key 'alt-x' };
-    assert_screen 'yast2_ldap_configuration_summary';
+    wait_screen_change { send_key 'alt-n' };
+    type_string 'dc=qa, dc=suse, dc=de';
+    # check advanced settings befor run test connection to ldap server
+    wait_screen_change { send_key 'alt-v' };
+    send_key 'ret';
+    # enter expert ldap settings
+    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings';
 
-    # finish ldap server configuration
-    wait_screen_change { send_key 'alt-f' };
-    wait_serial("yast2-auth-server-status-0") || die "'yast2 auth server' failed";
-    assert_screen 'yast2_console-finished';
+    ## expert ldap settings
+    # change replication sleep and time-out
+    wait_screen_change { send_key 'alt-p' };
+    type_string "990\n";
+    wait_screen_change { send_key 'alt-t' };
+    type_string "7\n";
 
-    # check ldap server status at first, a local ldap server is needed in the test case
-    systemctl "show -p ActiveState slapd.service | grep ActiveState=active";
+    # change to not use SSL or TLS
+    wait_screen_change { send_key 'alt-u' };
+    send_key 'up';
+    assert_screen 'yast2_samba-server_ldap_advanced_expert_settings_not-use-ssl';
+    wait_screen_change { send_key 'ret' };
+    wait_screen_change { send_key 'alt-o' };
+
+    # now run test connection
+    send_key 'alt-t';
+    assert_screen 'yast2_samba-server_ldap_test-connection';
+    wait_screen_change { send_key 'alt-o' };
 }
 
-sub gui_current {
+sub setup_samba {
+    script_run("yast2 samba-server; echo yast2-samba-server-status-\$? > /dev/$serialdev", 0);
+
     # samba-server configuration for SLE older than 15 or opensuse TW
     assert_screen([qw(yast2_samba_installation yast2_still_susefirewall2)], 60);
     if (match_has_tag 'yast2_still_susefirewall2') {
         send_key 'alt-c';
     }
+
     wait_screen_change { send_key 'alt-w' };
     for (1 .. 12) { send_key 'backspace'; }
     # give a new name for Workgroup
-    type_string 'QA-Workgroup';
+    type_string($samba_directives{workgroup});
     assert_screen 'yast2_samba-server_workgroup_new';
     send_key 'alt-n';
 
-    # select "Not a Domain Controller"
-    assert_screen 'yast2_samba_server_selection';
-    send_key 'alt-c';
+    ## Handle Domain Controller in sle12
+    if ((is_sle('<15')) || (is_leap('<15.0'))) {
```

Comment:
I guess we can remove wrapper brackets here (that's optional, but IMHO it's easier to read without them)